lib/Twig/Extensions/Node/Trans.php

```diff
@@ -41,6 +41,10 @@ public function compile(Twig_Compiler $compiler)
 
         $function = null === $this->getNode('plural') ? 'gettext' : 'ngettext';
 
+        if (null !== $this->getNode('domain')) {
```

Comment:
Actually I can hardly see the difference between how plural translations and domain translations are handled. Why should the domain one be handled at runtime while plurals are handled compile time. Both mutate the executed function, and both add values to the compile logic. So how does the two differ?